packages/components/upload/src/ajax.ts

```diff
@@ -70,8 +70,13 @@ export const ajaxUpload: UploadRequestHandler = (option) => {
   const formData = new FormData()
   if (option.data) {
     for (const [key, value] of Object.entries(option.data)) {
-      if (isArray(value) && value.length) formData.append(key, ...value)
-      else formData.append(key, value)
+      if (isArray(value) && value.length) {
+        for (const val of value) {
+          formData.append(key, val)
+        }
```

Comment:
The value type of `data` has no restrictions. If the value of a certain attribute is special, then this modification may cause the original behavior that is inconsistent.
```ts
// for example
const blob = new Blob(['text'], { type: 'text/plain' })
const data = {
  test: [blob, 'testname']
}
// origin 
formData.append('test', ...[data.test])

// changed
formData.append('test', data.test[0])
formData.append('test', data.test[1])
```
The above two writing effects seem to be inconsistent. 🤔

Reply:
you are right, value type is string | Blob, formData object value type is string or Blob
![image](https://github.com/element-plus/element-plus/assets/56245609/c4cd65df-925a-4b96-bb59-762f13cc15b2)
![image](https://github.com/element-plus/element-plus/assets/56245609/318f0497-7e75-4773-87ad-d19ed545baaf)